cylc/flow/parsec/jinja2support.py

```diff
@@ -186,11 +186,17 @@ def jinja2environment(dir_=None):
                     envnsp[fname] = getattr(module, fname)
 
     # Import WORKFLOW HOST USER ENVIRONMENT into template:
-    # (usage e.g.: {{environ['HOME']}}).
+    # (Usage e.g.: {{environ['HOME']}}).
     env.globals['environ'] = os.environ
+
     env.globals['raise'] = raise_helper
     env.globals['assert'] = assert_helper
 
+    # And all `CYLC_` variables (this will include CYLC_WORKFLOW environment
+    # variables exported by `cylc` commands before config file parsing.
+    for key, val in os.environ.items():
+        if key.startswith('CYLC_'):
+            env.globals[key] = val
     return env
```

Comment:
Note this will include `CYLC_VERSION` and `CYLC_ENV_NAME` if using the wrapper script.

I think `CYLC_VERSION` is already set elsewhere?

Reply:
Does that matter?

Reply:
Hmm, yeah the "truth" `CYLC_VERSION` is inserted there in `parsec.fileparse`.

I'll deal with this one tomorrow. And I'll move the env export into parse.fileparse to catch empy as well.

Reply:
Addressing ...

Reply:
I'm now not taking CYLC_VERSION or CYLC_ENV_NAME from the environment, as template variables.